### DISCUSSION TITLE
unable to funtion the swiper when following the Getting Started With Swiper Page

### DISCUSSION THREAD

**Discussion by neptunewong1998** [nolimits4web/swiper#5968] [Q&A] [answered]
i followed the [getting start pages](https://swiperjs.com/get-started#download-assets) in swipper.
and My swipper is located inside a grid-container.

i added the following code. however, the navigation seems doesn't work. can anyone tell me what I am missing??

```
    <script>

        var swiper = new Swiper('.swiper', {
            navigation: {
                nextEl: '.swiper-button-next',
                prevEl: '.swiper-button-prev',
            },
        });

    </script>

```
```
<link
  rel="stylesheet"
  href="https://cdn.jsdelivr.net/npm/swiper@8/swiper-bundle.min.css"
/>

<script src="https://cdn.jsdelivr.net/npm/swiper@8/swiper-bundle.min.js"></script>
```
html
``` 
                <div class="swiper">
                  <div class="swiper-wrapper">
                    <div class="swiper-slide">Slide 1</div>
                    <div class="swiper-slide">Slide 2</div>
                    <div class="swiper-slide">Slide 3</div>
                    <div class="swiper-slide">Slide 4</div>
                  </div>
                  <div class="swiper-button-next"></div>
                  <div class="swiper-button-prev"></div>
                </div>
```

i have done some search here. and found out that they both have difference solution
i tried to change the scripts, but this also not working.
```
    <script>
       Swiper.use([Navigation]);
        var swiper = new Swiper('.swiper', {
            navigation: {
                nextEl: '.swiper-button-next',
                prevEl: '.swiper-button-prev',
            },
        });

    </script>

```

**Comment by Arhell** [comment]
Try use https://github.com/nolimits4web/swiper/discussions/5839#discussioncomment-3068393

**Reply by neptunewong1998** [reply]
Thank you for ur reply.  i have changed my script to this however, it is still not working
```
    <script>
       Swiper.use([Navigation]);
        var swiper = new Swiper('.swiper', {
            navigation: {
                nextEl: '.swiper-button-next',
                prevEl: '.swiper-button-prev',
            },
        });

    </script>
```

**Reply by Arhell** [reply]
Try adding a custom class https://codesandbox.io/s/5vdwrg?file=/index.html

`mySwiper`

**Reply by neptunewong1998** [reply]
Sorry, i am a little bit confuse. You mean changing the class to the following code??
Do I need to change the scripts??

html
```
                <div class="swiper mySwiper">
                  <div class="swiper-wrapper">
                    <div class="swiper-slide">Slide 1</div>
                    <div class="swiper-slide">Slide 2</div>
                    <div class="swiper-slide">Slide 3</div>
                    <div class="swiper-slide">Slide 4</div>
                  </div>
                  <div class="swiper-button-next"></div>
                  <div class="swiper-button-prev"></div>
                </div>
 ```

**Reply by Arhell** [reply]
No, don't change your script, just add a new class to yours

```
  <script>

        var swiper = new Swiper('.NewClass', {
            navigation: {
                nextEl: '.swiper-button-next',
                prevEl: '.swiper-button-prev',
            },
        });

    </script>
```
```
                <div class="swiper NewClass">
                  <div class="swiper-wrapper">
                    <div class="swiper-slide">Slide 1</div>
                    <div class="swiper-slide">Slide 2</div>
                    <div class="swiper-slide">Slide 3</div>
                    <div class="swiper-slide">Slide 4</div>
                  </div>
                  <div class="swiper-button-next"></div>
                  <div class="swiper-button-prev"></div>
                </div>
```

**Reply by neptunewong1998** [reply]
i have tried the method and it seems it is still not working.
html
```
<div class="swiper mySwiper">
.....
```
scripts
```
Swiper.use([Navigation]);
var swiper = new Swiper('.mySwiper', {
....
or
Swiper.use([Navigation]);
var swiper = new Swiper('.swiper mySwiper', {
.....
```

**Reply by Arhell** [reply]
```
Swiper.use([Navigation]);
var swiper = new Swiper('.mySwiper'
```

**Reply by neptunewong1998** [reply]
yes I am currently using this. However, it is no functioning.
```
    Swiper.use([Navigation]);
    var swiper = new Swiper('.mySwiper', {
        navigation: {
            nextEl: '.swiper-button-next',
            prevEl: '.swiper-button-prev',
        },
    });
```

**Reply by Arhell** [reply]
Ok
interesting
use please https://cdn.jsdelivr.net/npm/swiper/swiper-bundle.min.js
and https://cdn.jsdelivr.net/npm/swiper/swiper-bundle.min.css

**Reply by neptunewong1998** [reply]
still no response. when updated to
```
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/swiper/swiper-bundle.min.css"
    />

<script src=" https://cdn.jsdelivr.net/npm/swiper/swiper-bundle.min.js"></script>
```
I can open all 4 links on chrome.

**Reply by Arhell** [reply]
Let's try another option
Please copy this all https://codesandbox.io/s/5vdwrg?file=/index.html

And see if it works or not, delete yours accordingly or comment it out

**Reply by neptunewong1998** [reply]
it can work.......... but how come when i plug into my code. it is not functioning.....

**Reply by Arhell** [reply]
And you installed everything like here? use npm? Do you have package.json?
https://swiperjs.com/get-started#installation

**Reply by neptunewong1998** [reply]
In the previous trial, I used the method "Use Swiper from CDN"
Do I still need to add anythings into my code??
what do u mean for package.json?

Actually I would like to use the method "Download assets". 
Do i still need to install other files?? I only dowload the file inside [ https://www.jsdelivr.com/package/npm/swiper](https://www.jsdelivr.com/package/npm/swiper)
and I have tried to change to the following code but it is still not working.
```
    <link
      rel="stylesheet"
      href="swiper/swiper-bundle.min.css"
    />

<script src=" swiper/swiper-bundle.min.js"></script>
```
Do I still need to install any others things??

**Reply by Arhell** [reply]
I just created index.html and pasted all the code from here https://codesandbox.io/s/5vdwrg?file=/index.html:0-2451 everything works

**Reply by neptunewong1998** [reply]
it also work for me. but when i embeded to my aspx file. it just no response.

**Reply by neptunewong1998** [reply]
thank you for ur help. I think I have solved the problem. The <script> must be placed within the <body>.

**Comment by neptunewong1998** [comment, marked as answer]
thank you for ur help. I think I have solved the problem. The <script> must be placed within the "body" tag .
